iroh/src/endpoint/quic.rs

```diff
@@ -0,0 +1,491 @@
+//! Exporting and encapsulating structs from quinn
+//!
+//! Co-locates all iroh-quinn exports.
+//!
+//! There are some structs that we use in particular ways, where we would like
+//! to limit or expand how those structs are used in iroh. By encapsulating them
+//! we can ensure the functionality needed to make iroh work.
+
+use std::{sync::Arc, time::Duration};
+
+use n0_error::{e, stack_error};
+pub(super) use quinn::TransportConfig;
+// Missing still: SendDatagram and ConnectionClose::frame_type's Type.
+pub use quinn::{
+    AcceptBi, AcceptUni, AckFrequencyConfig, ApplicationClose, Chunk, ClosedStream,
+    ConnectionClose, ConnectionError, ConnectionStats, MtuDiscoveryConfig, OpenBi, OpenUni,
+    PathStats, ReadDatagram, ReadError, ReadExactError, ReadToEndError, RecvStream, ResetError,
+    RetryError, SendDatagramError, SendStream, ServerConfig, StoppedError, StreamId, VarInt,
+    VarIntBoundsExceeded, WeakConnectionHandle, WriteError,
+};
+pub use quinn_proto::{
+    FrameStats, IdleTimeout, TransportError, TransportErrorCode, UdpStats, Written,
+    congestion::{Controller, ControllerFactory},
+    crypto::{
+        AeadKey, CryptoError, ExportKeyingMaterialError, HandshakeTokenKey,
+        ServerConfig as CryptoServerConfig, UnsupportedVersion,
+    },
+};
+
+use crate::magicsock::{HEARTBEAT_INTERVAL, MAX_MULTIPATH_PATHS, PATH_MAX_IDLE_TIMEOUT};
+
+/// Parameters governing the core QUIC state machine
+///
+/// Default values should be suitable for most internet applications. Applications protocols which
+/// forbid remotely-initiated streams should set `max_concurrent_bidi_streams` and
+/// `max_concurrent_uni_streams` to zero.
+///
+/// In some cases, performance or resource requirements can be improved by tuning these values to
+/// suit a particular application and/or network connection. In particular, data window sizes can be
+/// tuned for a particular expected round trip time, link capacity, and memory availability. Tuning
+/// for higher bandwidths and latencies increases worst-case memory consumption, but does not impair
+/// performance at lower bandwidths and latencies. The default configuration is tuned for a 100Mbps
+/// link with a 100ms round trip time.
+///
+/// In iroh, the config has some specific default values that make iroh's holepunching work
+/// well with QUIC multipath. Adjusting those settings may error if the given values
+/// are out of scope with what is optimal for iroh.
+///
+/// Look at the following methods for more details:
+/// - [`QuicTransportConfig::keep_alive_interval`]
+/// - [`QuicTransportConfig::default_path_keep_alive_interval`]
+/// - [`QuicTransportConfig::default_path_max_idle_timeout`]
+/// - [`QuicTransportConfig::max_concurrent_multipath_paths`]
+/// - [`QuicTransportConfig::set_max_remote_nat_traversal_addresses`]
+#[derive(Debug)]
+pub struct QuicTransportConfig(pub(crate) TransportConfig);
```

Comment:
nit: I think this would be more readable:

```suggestion
pub struct QuicTransportConfig(pub(crate) quinn::TransportConfig);
```